### ISSUE TITLE
rboot_write_flash : writing final extra bytes

### ISSUE THREAD

**Issue by hreintke** [SmingHub/SmingRTOS#101]
@raburton : 
I am implementing a "FlashStream" as a way to write raw data to flash without using the filesystem.
Usage within Sming will be the possibility to push (OTA) application/ spiffs FS  using HTTP upload, but of course can then be used for any other purpose.
As function to write to the flash I intend to use the rboot_write_flash, as it already takes care of erasing and alignment of 4 bytes.

Checking the functionality, I did not see the possibility to write as a last action the remaining "extra bytes" from the rboot_write_status.
Did I overlook this or needs this still to be implemented ?

**Comment by raburton** [comment]
Sorry for slow reply, been busy lately and currently have no internet access at home! You are correct that up to 3 trailing bytes could be lost. Not an issue for roms or spiffs, which are always multiples of 4 bytes, but I should probably add a finalise call to make sure any remaining bytes are written out in case people want to write anything that isn't. I'll get on this once I have internet access again...

**Comment by hreintke** [comment]
No hurry. I already used another solution.
From the work on Spiffs I remembered that the sming flashwrite routine take care of first reading some bytes before writing to a "non 4 byte aligned" location.
On top of that I wrote a "flashmem_erase_write" function which erases a sector when first writing to that sector. That solves the "last bytes" issue and doesn't need the separate structure in the write routine. On the cost of sometimes an additional read before write. Which is acceptable.
